pyiron/atomistics/master/elastic.py

```diff
@@ -0,0 +1,385 @@
+            already specified)
+
+    The default input parameters might not be chosen adequately. If you have a large
+    computation power, increase `min_num_measurements`. At the same time, make
+    sure to choose an orientation which maximizes the number symmetry operations.
+    Also if the child job does not support pressure, better increase the number
+    of measurements.
+    """
+    def __init__(self, project, job_name):
+        super().__init__(project, job_name)
+        self.__name__ = "ElasticTensor"
+        self.__version__ = "0.1.0"
+
+        self.input["min_num_measurements"] = (11, "minimum number of samples to be taken")
+        self.input["min_num_points"] = (105, "minimum number of data points"
+            + "(number of measurements will be min_num_points/len(rotations))")
+        self.input["max_strain"] = (
+            0.01,
+            "relative volume variation around volume defined by ref_ham",
+        )
+        self.input['polynomial_order'] = 2
+        self.input['additional_points'] = 0
+        self.input['strain_matrices'] = []
+        self.input['use_symmetry'] = True
+        self.input['rotations'] = []
+        self.input['normalize_magnitude'] = False
+        self.input['use_elements'] = (True, 'whether or not consider chemical elements for '
+                                            + 'the symmetry analysis. Could be useful for SQS')
+        self.input['fit_first_order'] = False
+        self._job_generator = ElasticJobGenerator(self)
+
+    @property
+    def _number_of_measurements(self):
+        return int(max(
+            self.input['min_num_measurements'],
+            np.ceil(self.input['min_num_points']/max(len(self.input['rotations']), 1))
+        ))
+
+    def _get_rotation_matrices(self):
+        rotations = self.ref_job.structure.get_symmetry(use_elements=self.input['use_elements'])['rotations']
+        _, indices = np.unique(np.round(rotations, 6), axis=0, return_inverse=True)
+        rotations = rotations[np.unique(indices)]
+        self.input['rotations'] = rotations.tolist()
+
+    def _create_strain_matrices(self):
+        if self.input['use_symmetry'] and len(self.input['rotations'])==0:
+            self._get_rotation_matrices()
+        eps_lst = np.random.random((int(self._number_of_measurements), 3, 3))-0.5
+        eps_lst *= self.input['max_strain']
+        eps_lst += np.einsum('nij->nji', eps_lst)
+        self.input['strain_matrices'] = get_strain(max_strain=self.input['max_strain'],
+                                                   n_set=self._number_of_measurements,
+                                                   polynomial_order=self.input['polynomial_order'],
+                                                   additional_points=self.input['additional_points'],
+                                                   normalize=self.input['normalize_magnitude']).tolist()
+
+    def validate_ready_to_run(self):
+        super().validate_ready_to_run()
```

Comment:
I'm a little torn whether we should be doing these in setters rather than all the way at `validate_ready_to_run` -- I don't think you need to change anything here! It's just a general question I have.

I really like all the checks and find the error messages clear :)

Reply:
Yeah that's true. It's just this `input` is inherited from somewhere deep down and I didn't really want to look into it too much. But it's true that for example in the case of `polynomial_order` it makes more sense to do it in the setter. When we switch to `InputList` I'll do it at the same time.